src/main/groovy/pl/allegro/tech/build/axion/release/infrastructure/di/Context.groovy

```diff
@@ -31,7 +31,20 @@ class Context {
 
         instances[ScmRepository] = scmRepository
         instances[VersionService] = new VersionService(new VersionResolver(scmRepository))
+    }
+
+    public Context(Properties rules, ScmRepository scmRepository, ScmProperties scmProperties, File projectRoot, LocalOnlyResolver localOnlyResolver, VersionConfig versionConfig) {
```

Comment:
the idea behind having `Context` and `GradleAwareContext` classes separate was to isolate any Gradle-dependent code from plugin domain. The isolation is not perfect, but i would like to try and keep it this way. `VersionConfig` object in my view is part of Gradle config (it is mutable etc), and so it should not be imported by `Context` class. Can you change it? I know it will make constructor uglier, so maybe some `Properties` object is needed.

Reply:
also is old constructor needed? why not keep only one, extended constructor?

Reply:
Please would you mind confirming your view on what `Context` should do - it's a class to encapsulate all the Git-specific classes/tools etc, right?  So `Context` should have some sort of properties (`MonoRepoProperties`) that are used to process mono-repos, and this class would be created via a similar static method to the call to `LocalOnlyResolverFactory.create(...)`?

e.g. 
`MonoRepoProperties.createFromConfig(MonoRepoConfig config)
`
where `MonoRepoconfig` is a class contained within `VersionConfig`?

Reply:
`Context` is a class to build the whole core environment - meaning version reading & resolution. Core logic should have nothing to do with Gradle.

Each `*Config` file is transformed into `*Properties` using a separate `*Factory`. Factory transforms Gradle domain objects (`*Config`) into core domain (`*Properties`) while handing cmd args and env overwrites. `*Factory` classes belong to Gradle, as they are aware of Gradle.

So, in general, the second answer :)

Reply:
Amended.